### PR TITLE
Enabled SA1600 (Elements must have docs) - Added required comments

### PR DESCRIPTION
**Pull request by alistairjevans**
I've turned on S1600 in the analyzer ruleset and added the missing comments. There were only about 200 of them...

I've also added a suppression file for the test projects so they don't need docs, because the ruleset can't vary between projects.

### CODE REVIEW

**Comment by alistairjevans** [comment]
If I had a dollar for every time I've absent-mindedly removed a method because CodeLens tells me it's unused, for everything to break horribly.

File: build/Analyzers.ruleset [hunk 1]
```diff
@@ -1,61 +1,33 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="Autofac Analyzer Rules" Description="Analyzer rules for Autofac assemblies." ToolsVersion="14.0">
-  <!-- https://github.com/dotnet/roslyn/blob/master/docs/compilers/Rule%20Set%20Format.md -->
+<RuleSet Name="Autofac Analyzer Rules" Description="Analyzer rules for Autofac assemblies." ToolsVersion="16.0">
```
Comment (tillig) [inline build/Analyzers.ruleset:2]: I think the comments about the rules were helpful. Can we revert this?

Comment (tillig) [reply build/Analyzers.ruleset:2]: (Keep the updated ToolsVersion, but put the comments back, I mean.)

File: src/Autofac/Features/GeneratedFactories/GeneratedFactoryRegistrationSource.cs [hunk 2]
```diff
@@ -32,6 +32,9 @@
 
 namespace Autofac.Features.GeneratedFactories
 {
+    /// <summary>
+    /// Registration source for generated factory methods (i.e. when resolving <![CDATA[Func<TService>]]> or some variant).
```
Comment (tillig) [inline src/Autofac/Features/GeneratedFactories/GeneratedFactoryRegistrationSource.cs:36]:
```suggestion
    /// Registration source for generated factory methods (i.e. when resolving <see cref="Func{T}"/> or some variant).
```

File: test/TestSuppressions.cs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+﻿// Suppressions file for test projects.
```
Comment (tillig) [inline test/TestSuppressions.cs:1]: Is this better than `NoWarn` on SA1600? (I don't have a real preference, just curious if you find one way to be better than the other.)

In other projects I've done, I actually have two different rulesets - one for production code, one for tests. The test one is less stringent. Maybe something to consider in the future.

Comment (alistairjevans) [reply test/TestSuppressions.cs:1]: I just did it like this so I could share it, but perhaps it's better to put the NoWarn in a Directory.props file in the tests folder?

File: build/Analyzers.ruleset (hunk 1 above)
Comment (alistairjevans) [reply build/Analyzers.ruleset:2]: Yeah, I can do that; Visual Studio must have borked them.

File: test/TestSuppressions.cs (hunk 3 above)
Comment (tillig) [reply test/TestSuppressions.cs:1]: Meh. It's fine for now. Some later point I might put two different ruleset files in there for better control, but I'm not going to be a stickler over it.

File: src/Autofac/Features/GeneratedFactories/GeneratedFactoryRegistrationSource.cs (hunk 2 above)
Comment (alistairjevans) [reply src/Autofac/Features/GeneratedFactories/GeneratedFactoryRegistrationSource.cs:36]: Yeah, I don't why I didn't just do that...

File: test/TestSuppressions.cs (hunk 3 above)
Comment (alistairjevans) [reply test/TestSuppressions.cs]: Whoops, too late!

**Pull request merged**